app/assets/javascripts/thyme/thyme_player.js

```diff
@@ -0,0 +1,212 @@
+  function strokeColorFunc(annotation) {
+    return 'black';
+  }
+  function sizeFunc(annotation) {
+    return false;
+  }
+  function onClick(annotation) {
+    annotationArea.update(annotation);
+    annotationArea.show();
+    $('#caption').hide();
+  }
+  function onUpdate() {
+    /* update might change the annotation which is currently shown in the
+       annotation area -> find the updated annotation in the annotation array
+       and update the area. */
+    if (annotationArea.annotation != null) {
+      const id = annotationArea.annotation.id;
+      annotationArea.update(AnnotationManager.find(id));
+    }
+  }
+  const annotationManager = new AnnotationManager(colorFunc, strokeColorFunc, sizeFunc,
+                                                  onClick, onUpdate, isValid);
+  thymeAttributes.annotationManager = annotationManager;
+  // onShow and onUpdate definition for the annotation area
+  function onShow() {
+    iaButton.plus();
+    annotationManager.updateMarkers();
+  }
+  function onHide() {
+    iaButton.minus();
+    annotationManager.updateMarkers();
+    resizeContainer();
+  }
+  annotationArea.onShow = onShow;
+  annotationArea.onHide = onHide;
+
+  // Update annotations after submitting the annotations form
+  $(document).on('click', '#annotation-modal-submit-button', function() {
+    /* NOTE:
+       Updating might take some time on the backend,
+       so I added a slight delay.
+       I couldn't think of an easy way to let the script
+       wait for the update to complete (as with the delete button),
+       but it might be possible! */
+    setTimeout(function() {
+      annotationManager.updateAnnotations();
+    }, 500);
+  });
+
+  // Update annotations after deleting an annotation
+  $(document).on('click', '#delete-button', function() {
+    const annotationId = Number(document.getElementById('annotation_id').textContent);
+    $.ajax(Routes.annotation_path(annotationId), {
+      type: 'DELETE',
+      dataType: 'json',
+      data: {
+        annotationId: annotationId
+      },
+      success: function() {
+        annotationManager.updateAnnotations();
+        // close and open again = show normal IA
+        iaButton.minus();
+        iaButton.plus();
+      }
+    });
+  });
+
+
+
+  /*
+    CHAPTERS & METADATA MANAGER
+   */
+  const iaBackButton = new IaBackButton('back-button', 'chapters');
+  iaBackButton.add();
+  const chapterManager  = new ChapterManager('chapters', iaBackButton);
+  const metadataManager = new MetadataManager('metadata');
+  thymeAttributes.chapterManager  = chapterManager;
+  thymeAttributes.metadataManager = metadataManager;
+  chapterManager.load();
+  metadataManager.load();
+
+
+
+  /*
+    INTERACTIVE AREA
+   */
+  iaButton.toHide = [$('#caption'), annotationArea];
+  iaButton.add();
+  (new IaCloseButton('ia-close', iaButton)).add();
+  //TODO
+  //const interactiveArea = new InteractiveArea();
+  //thymeAttributes.interactiveArea = interactiveArea;
```

Comment:
Good catch! I wanted to create an InteractiveArea class, because I also have created an AnnotationArea class and in theory this would lead to a better code structure. However, in practise the interactive area is far more complicated than the annotation area and as far as I tried, the separated class only made things more complicated.
In future, I think it would improve the code structure if we think again about this carefully, but I'd like to postpone this to a future release.